File: python/meteroid/models/product.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .product_family_id import ProductFamilyId
from .product_fee_structure import ProductFeeStructure
from .product_fee_type_enum import ProductFeeTypeEnum
from .product_id import ProductId


@dataclasses.dataclass
class Product(BaseModel):
    catalog: bool

    created_at: datetime

    fee_structure: ProductFeeStructure

    fee_type: ProductFeeTypeEnum

    id: ProductId

    name: str

    product_family_id: ProductFamilyId

    archived_at: t.Optional[datetime] = None

    description: t.Optional[str] = None

File: go/product.go
// this file is @generated
package meteroid

import "time"

type Product struct {
	// RFC 3339 timestamp.
	ArchivedAt *time.Time `json:"archived_at,omitempty"`

	Catalog bool `json:"catalog"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Description *string `json:"description,omitempty"`

	FeeStructure ProductFeeStructure `json:"fee_structure"`

	FeeType ProductFeeTypeEnum `json:"fee_type"`

	Id ProductId `json:"id"`

	Name string `json:"name"`

	ProductFamilyId ProductFamilyId `json:"product_family_id"`
}
